datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -126,14 +127,18 @@ impl ParquetExec {
                     }
                 }
             })
-            .and_then(|pruning_predicate| {
-                // If the pruning predicate can't prune anything, don't try
-                if pruning_predicate.allways_true() {
+            .filter(|p| !p.allways_true());
+
+        let page_pruning_predicate = predicate.as_ref().and_then(|predicate_expr| {
+            match PagePruningPredicate::try_new(predicate_expr, file_schema.clone()) {
+                Ok(pruning_predicate) => Some(Arc::new(pruning_predicate)),
+                Err(e) => {
+                    debug!("Could not create page pruning predicate for: {}", e);
```

Comment:
I think it would be worth including the `predicate_expr` in this message too

```suggestion
                    debug!("Could not create page pruning predicate for '{}': {}", predicate_expr, e);
```

Reply:
Proposed change in https://github.com/apache/arrow-datafusion/pull/4736